pkg/controller/operators/catalog/operator.go

```diff
@@ -34,13 +34,18 @@ const (
 //for test stubbing and for ensuring standardization of timezones to UTC
 var timeNow = func() metav1.Time { return metav1.NewTime(time.Now().UTC()) }
 
+type sourceKey struct {
+	name      string
+	namespace string
+}
+
 // Operator represents a Kubernetes operator that executes InstallPlans by
 // resolving dependencies in a catalog.
 type Operator struct {
 	*queueinformer.Operator
 	client            versioned.Interface
 	namespace         string
-	sources           map[string]registry.Source
+	sources           map[sourceKey]registry.Source
```

Comment:
Use complex `struct` which contains both name and namespace as map key.

Reply:
:+1: for compound keys